src/sdc11073/xml_types/xml_structure.py

```diff
@@ -828,15 +946,28 @@ def update_xml_value(self, instance, node):
 
 
 class ContainerProperty(_ElementBase):
-    """ a value that has "mk_node" and cls.from_node methods"""
-
-    def __init__(self, sub_element_name, value_class, cls_getter, ns_helper, is_optional=False):
+    """ContainerProperty is a property that supports xsi:type information from xml and instantiates value accordingly."""
+
+    def __init__(self, sub_element_name: etree_.QName | None,  # Noqa: PLR0913
+                 value_class: Any,
+                 cls_getter: Callable,
```

Comment:
type could be extended to (based on documentation): `Callable[[etree_.QName], typing.Type]`